trinity/main.py

```diff
@@ -294,6 +306,7 @@ def exit_because_ambigious_filesystem(logger: logging.Logger) -> None:
     await sigint_received.wait()
     try:
         await service_to_exit.cancel()
+        service_to_exit._executor.shutdown(wait=True)
```

Comment:
Maybe this should be in `BaseService`'s `cancel()`.

Reply:
Hah, right, makes sense...

Reply:
Which btw, and not really related to this PR:

I've been thinking about mapping out our service hierarchy sometime soon.  I think it'd be really useful to have a graph that shows the hierarchy of services as well as the peer subscriptions.  Otherwise, visualizing that stuff takes a **lot** of domain knowledge.

Reply:
I suppose we could *generate* this service hierarchy for display at runtime, which could be neat and not require maintenance as we add more services.

Reply:
Maybe we could do this as a plugin which gathered this information from the events that get broadcast over the event bus.  Subscriptions and running child services both strike me as reasonable events to be fired off and with that it'd be pretty easy to scrape the application hierarchy.

Alternatively, we would need to add some extra API surface area to the Service and subscription API to make this data available.  Gonna put this on the back-burner for now, but I like the idea.